src/index.js

```diff
@@ -1,32 +1,152 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
-
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
-
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
-
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
+const response_then = document.querySelector('.response_then');
+const response_asyncawait = document.querySelector('.response_asyncawait');
+const API = "https://rickandmortyapi.com/api/character/";
+
+function fetchData(url_api, data) {
+  const xhttp = new XMLHttpRequest();
+  xhttp.responseType = 'json';
+
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          resolve({
+            response: xhttp.response,
+            prevData: data
+          });
+        } else return reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```